File: system-test/java/src/main/java/com/optivem/shop/dsl/port/then/steps/ThenOrder.java
package com.optivem.shop.dsl.port.then.steps;

import com.optivem.shop.dsl.driver.port.shop.dtos.OrderStatus;
import com.optivem.shop.dsl.port.then.steps.base.ThenStep;

public interface ThenOrder extends ThenStep<ThenOrder> {
    ThenOrder hasSku(String expectedSku);

    ThenOrder hasQuantity(int expectedQuantity);

    ThenOrder hasUnitPrice(double expectedUnitPrice);

    ThenOrder hasTotalPrice(double expectedTotalPrice);

    ThenOrder hasTotalPrice(String expectedTotalPrice);

    ThenOrder hasStatus(OrderStatus expectedStatus);

    ThenOrder hasTotalPriceGreaterThanZero();

    ThenOrder hasOrderNumberPrefix(String expectedPrefix);
}

File: system-test/typescript/src/testkit/dsl/port/then/steps/then-order.ts
export interface ThenOrder extends PromiseLike<void> {
  and(): this;
  hasSku(expectedSku: string): this;
  hasQuantity(expectedQuantity: number): this;
  hasUnitPrice(expectedUnitPrice: number): this;
  hasBasePrice(basePrice: string | number): this;
  hasSubtotalPrice(subtotalPrice: string | number): this;
  hasTotalPrice(totalPrice: string | number): this;
  hasStatus(expectedStatus: string): this;
  hasTotalPriceGreaterThanZero(): this;
  hasOrderNumberPrefix(expectedPrefix: string): this;
  hasDiscountRate(expectedDiscountRate: number): this;
  hasDiscountAmount(expectedDiscountAmount: string | number): this;
  hasAppliedCouponCode(expectedCouponCode: string | null): this;
  hasAppliedCoupon(expectedCouponCode?: string): this;
  hasTaxRate(expectedTaxRate: string | number): this;
  hasTaxAmount(expectedTaxAmount: string | number): this;
}
